### PR TITLE
arm: stm32f7: add support for the stm32f765xx soc

### PR DESCRIPTION
**Pull request by rahul-arasikere**
The stm32f765xx series is similar to stm32f767xx series but doesn't include an LCD-TFT peripheral. I have updated the device tree files to have a separate one for the stm32f765 which is then included by the stm32f767 to add lcd-tft display controller node.

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Looks or but some re-ordering required on the commits.

Alos, as reported by CI, your pull request does exctly conform with Zephyr contribution guidelines. For reference, I'd advise you to check:
- [Contribution guidelines on docs.zephyrproject.org](https://docs.zephyrproject.org/latest/contribute/guidelines.html)
- [Great presentation dealing with "the best way to get your patches merged"](https://youtu.be/mnVNLM7ynMQ) and the related [Slides](https://static.sched.com/hosted_files/zephyr2022/a3/HOWTO_Get_Your_Zephyr_Patches_Merged_ZDS_2022.pdf)
- Coding Guidelines bot results reported below.

Please:
- Check your commit messages. It should conform to the following template:
```
<dir>: <dir>: <Title summarizing the change>

<A description useful to the reader>

Signed-off-by: foo <foo@bar.baz>
```
- Remove the merge commit(s)
- ...

File: dts/arm/st/f7/stm32f765.dtsi [hunk 1]
```diff
@@ -25,6 +25,26 @@
     soc {
         compatible = "st,stm32f765", "st,stm32f7", "simple-bus";
 
+        pinctrl: pin-controller@40020000 {
```
[the remark this answers is not in the document]
Comment (rahul-arasikere) [reply dts/arm/st/f7/stm32f765.dtsi]: Done.

File: soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f765xx [hunk 2]
```diff
@@ -9,6 +9,6 @@ config SOC
 	default "stm32f765xx"
 
 config NUM_IRQS
-	default 98
+	default 110
```
[the remark this answers is not in the document]
Comment (rahul-arasikere) [reply soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f765xx]: Done.

[the remark this answers is not in the document]
Comment (erwango) [reply soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f765xx]: Still in the wrong commit :)

**Comment by erwango** [comment]
One last issue: https://github.com/zephyrproject-rtos/zephyr/actions/runs/6026405699/job/16349554945?pr=62055

To update the author name, see https://www.git-tower.com/learn/git/faq/change-author-name-email

File: soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f765xx (hunk 2 above)
[the remark this answers is not in the document]
Comment (rahul-arasikere) [reply soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f765xx]: I believe third time is the charm and should be good to go now.

**Review by erwango — APPROVED** [review]
Approved!
Thanks

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**